### PR TITLE
🧑‍⚕️Add wordpress for BMES.

### PR DESCRIPTION
**Pull request by SohamG**
- Add BMES website https://convergehealth.cc
- Revert #95 to allow cert-manager to work again
- Deploy wordpress for BMES.
- ???
- Profit

### CODE REVIEW

File: kubernetes/argocd/stacks/bmes-wordpress/main.yml [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+            - name: http
+              containerPort: 80
+              protocol: TCP
+          volumeMounts:
+            - name: wp-data
+              mountPath: /var/www/html
+          env:
+            - WORDPRESS_DB_HOST: localhost
+            - WORDPRESS_DB_USER: root
+            - WORDPRESS_DB_PASSWORD: notASecret
```
Comment (clee231) [inline kubernetes/argocd/stacks/bmes-wordpress/main.yml:75]: No credentials in this codebase, please! :pray:

File: kubernetes/argocd/stacks/bmes-wordpress/main.yml [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+  entryPoints:
+    - websecure
+    - web
+  routes:
+    - kind: Rule
+      match: "Host(`convergehealth.cc`) || Host(`www.convergehealth.cc`)"
+      services:
+        - kind: Service
+          name: bmes # Perhaps move to new file.
+          namespace: bmes
```
Comment (clee231) [inline kubernetes/argocd/stacks/bmes-wordpress/main.yml:125]: Did you mean `bmes-wordpress`?

File: kubernetes/argocd/stacks/bmes-wordpress/main.yml [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+spec:
+  entryPoints:
+    - websecure
+    - web
+  routes:
+    - kind: Rule
+      match: "Host(`convergehealth.cc`) || Host(`www.convergehealth.cc`)"
+      services:
+        - kind: Service
+          name: bmes # Perhaps move to new file.
```
Comment (clee231) [inline kubernetes/argocd/stacks/bmes-wordpress/main.yml:124]: On line 97 above, you name the service `wordpress`, not `bmes` as you define in this IngressRoute.

**Review by clee231 — CHANGES_REQUESTED** [review]

File: kubernetes/argocd/stacks/bmes-wordpress/certs.yml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: bmes-tls
+spec:
+  dnsNames:
+    - convergehealth.cc
```
[the remark this answers is not in the document]
Comment (SohamG) [reply kubernetes/argocd/stacks/bmes-wordpress/certs.yml:8]: I fixed this

File: kubernetes/argocd/stacks/bmes-wordpress/main.yml [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+          persistentVolumeClaim:
+            claimName: database-pvc
+      containers:
+        - name: wordpress-db
+          image: "docker.io/mysql"
+          volumeMounts:
+            - name: wp-db
+              mountPath: /var/lib/mysql
+          env:
+            - MYSQL_ROOT_PASSWORD: notASecret
```
[the remark this answers is not in the document]
Comment (SohamG) [reply kubernetes/argocd/stacks/bmes-wordpress/main.yml]: the DB is not accessible from anywhere, its a multi container pod. Hence its not a secret

**Comment by SohamG** [comment]
NOTES:
- I couldn't get the DB container to create the user/db/password etc on its own thru env variables. Tried Docker.io/mysql and then linuxserver/mariadb. I had to shell in to the container and make the users manually. I did make sure to delete the volumes when i changed init-only vars, but it just did not like them. maybe fix some day.

- TLS Resolves now

File: kubernetes/argocd/stacks/common/projects.yml [hunk 6]
```diff
@@ -81,4 +81,23 @@ spec:
...
+  destinations:
+  - namespace: 'bmes-wordpress'
+    server: '*'
+  sourceRepos:
+  - 'git@github.com:acm-uic/IaC.git'
+  - 'https://acm.cs.uic.edu/helm'
+  roles:
+  # A role which provides read-only access to all applications in the project
+  - name: read-only
+    description: Read-only privileges to sparkhacks
```
[the remark this answers is not in the document]
Comment (SohamG) [reply kubernetes/argocd/stacks/common/projects.yml:103]: argh

[the remark this answers is not in the document]
Comment (SohamG) [reply kubernetes/argocd/stacks/common/projects.yml]: fixed

**Review by clee231 — APPROVED** [review]

**Pull request merged**